src/uthash.h

```diff
@@ -177,24 +193,45 @@ do {
 
 #define HASH_MAKE_TABLE(hh,head)                                                 \
 do {                                                                             \
+  int _mem_ok = 1;                                                               \
   (head)->hh.tbl = (UT_hash_table*)uthash_malloc(sizeof(UT_hash_table));         \
-  if (!(head)->hh.tbl) {                                                         \
+  if (!((head)->hh.tbl)) {                                                       \
+    if (HASH_NOMEM_OK) {                                                         \
+      break;                                                                     \
+    }                                                                            \
     uthash_fatal("out of memory");                                               \
```

Comment:
Well, that was the whole point. If the client wants to use exit/jmp, they should not define `HASH_NOMEM_OK`, and they will get what they want. `HASH_NOMEM_OK` explicitly turns on the mode where error handling is to be done by checking `hh.mem_failed` after a hash write operation. This should cover all 3 cases, and I don't see a point when you want to mix them together somehow. Doing exit/jmp will prevent `mem_failed` to be set up properly, it will also prevent clean up (you really really don't want to use `longjmp` or `exit_thread` just because of that). Doing anything else but jmp/exit will be questionable because it carries no information on where the failure occurred. You might as well deal with it by checking `mem_failed` after an operation.